Sync/SyncTelemetryUtils.swift

```diff
@@ -468,6 +468,39 @@ public class GleanSyncOperationHelper {
         GleanMetrics.Pings.shared.tempSync.submit()
     }
 
+    public func reportTelemetry(_ result: MozillaAppServices.SyncResult) {
+        guard let json = result.telemetryJson,
+              let telemetry = try? RustSyncTelemetryPing.fromJSONString(
+                jsonObjectText: json
+              ) else {
+            return
+        }
+
+        for sync in telemetry.syncs {
+            _ = GleanMetrics.Sync.syncUuid.generateAndSet()
+
+            if let failureReason = sync.failureReason {
+                GleanMetrics.Sync.failureReason[String(describing: failureReason)].add()
+            }
+
+            for engine in sync.engines {
+                if engine.failureReason == nil {
+                    self.recordRustSyncEngineStats(engine.name,
+                                                   incoming: engine.incoming,
+                                                   outgoing: engine.outgoing)
+                } else {
+                    self.recordSyncEngineFailure(
```

Comment:
Good catch because, yeah, I think this would be an issue. I meant to set up the glean debug viewer but got sidetracked by breaking up the PRs. I'll check and see what I get but I'm pretty sure I'll need to cast what's returned into the expected failure reasons.